controllers/metal3.io/baremetalhost_controller.go

```diff
@@ -1547,10 +1582,20 @@ func (r *BareMetalHostReconciler) handleDataImageActions(prov provisioner.Provis
 
 	attachedURL := dataImage.Status.AttachedImage.URL
 
-	if deleteDataImage {
-		info.log.Info("DataImage requested for deletion")
+	if deleteDataImage || detachDataImage {
+		detachReason := ""
+		if deleteDataImage {
+			detachReason = "Delete DataImage"
+		}
+		if isBMHAvailable {
+			detachReason = "BMH Deprovision finished, node in available state"
```

Comment:
Are we going to leave a stale DataImage object in this case? If yes, it will behave unpredictably on the next deployment because it will still have a non-empty URL.

Reply:
Good point, I will also set the `spec.url: ""`.